### PR TITLE
add `dynsim_kwargs` attr for `System`

### PR DESCRIPTION
**Pull request by yalinli2**
As we discussed, I added `dynsim_kwargs` as an attribute to `System`, All qsdsan's tests and biosteam's qsdsan test passed locally.

@yoelcortes , looks like tests are failing because you are using walrus operators that doesn't exist for Python 3.7, maybe you should change the Python version in README/test env to 3.8/3.9, with the remote tests failing almost all the time we aren't able to tell whether it's because our updates or other problems. Thanks!

### CODE REVIEW

File: biosteam/_system.py [hunk 1]
```diff
@@ -1624,54 +1628,70 @@ def clear_state(self):
...
-        sample_id : int or str
-            If `export_state_to` is provided with a valid path with extension
-            '.xlsx' or '.xls' , sample_id will be used as the sheet name.
-        kwargs : dict
-            Other keyword arguments that will be passed to ``solve_ivp``
-            or ``odeint``. Must contain t_span for ``solve_ivp`` or t for ``odeint``.
+        dynsim_kwargs : dict
+            Dynamic simulation keyword arguments, could include:
+
+                t_span : Iterable(float)
```
Comment (joyxyz1994) [inline biosteam/_system.py:1640]: might be better to just specify `2-tuple` in case an iterable of the wrong length is given

Comment (yalinli2) [reply biosteam/_system.py:1640]: It's kind of in the next line, but I'll change it to "tuple(float, float)"
> Interval of integration (t0, tf).

File: biosteam/_system.py [hunk 2]
```diff
@@ -1624,54 +1628,70 @@ def clear_state(self):
...
+            # Load initial states
             self._converge()
             y0, idx, nr = self._load_state()
             dydt = self.DAE
-            self.scope.sol = sol = solve_ivp(fun=dydt, y0=y0, **kwargs)
-            if sol.status == 0: print('Simulation completed.')
-            else: print(sol.message)
-            self._write_state()
+            dk['y0'] = y0
+            dk['dydt'] = dydt
```
[the remark this answers is not in the document]
Comment (yalinli2) [reply biosteam/_system.py:1686]: My original intention is that we can run `solve_ivp` without going through this compiling again, I guess I didn't realize that `dydt` is just `self.DAE`, can I use the following:
```python
            y0, idx, nr = self._load_state()
            dk['y0'] = y0
            # Integrate
            self.scope.sol = sol = solve_ivp(fun=self.DAE, y0=y0, **dk_cp)
```

File: biosteam/_system.py [hunk 3]
```diff
@@ -1624,54 +1628,70 @@ def clear_state(self):
...
+                    The solver starts with t=t0 and integrates until it reaches t=tf.
+                state_reset_hook: str or callable
+                    Hook function to reset the cache state between simulations
+                    for dynamic systems).
+                    Can be "reset_cache" or "clear_state" to call `System.reset_cache`
+                    or `System.clear_state`, or None to avoiding resetting.
+                export_state_to: str
+                    If provided with a path, will save the simulated states over time to the given path,
+                    supported extensions are ".xlsx", ".xls", "csv", and "tsv".
+                sample_id : int or str
```
[the remark this answers is not in the document]
Comment (yalinli2) [reply biosteam/_system.py:1651]: @joyxyz1994 I'm not entirely sure if I know what the problem is (or how `model.evaluate` should be updated, can you make the update on the dyn_sim branch? Thanks!

File: biosteam/_system.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (joyxyz1994) [reply biosteam/_system.py:1686]: Yep

File: biosteam/_system.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (joyxyz1994) [reply biosteam/_system.py:1651]: I checked the codes in `model.evaluate()`. It should be fine as it is now. You can simply remove `sample_id` from the documentation.

[the remark this answers is not in the document]
Comment (yalinli2) [reply biosteam/_system.py:1651]: sounds good, I updated the documentation accordingly

[the remark this answers is not in the document]
Comment (joyxyz1994) [reply biosteam/_system.py:1651]: Thanks!

**Pull request merged**